src/tasks/reddit.py

```diff
@@ -0,0 +1,202 @@
+"""Task definitions for reddit."""
+import codecs
+import logging as log
+import os
+
+from allennlp.training.metrics import Average
+
+from allennlp.data import Instance, Token
+from allennlp.data.fields import TextField, LabelField, MetadataField
+
+from ..utils.utils import process_sentence, truncate
+
+from typing import Iterable, Sequence, List, Dict, Any, Type
+
+from .tasks import RankingTask, PairClassificationTask
+from .tasks import sentence_to_text_field
+from .registry import register_task
+
+@register_task('reddit', rel_path='Reddit_2008/')
+@register_task('reddit_dummy', rel_path='Reddit_2008_TestSample/')
+@register_task('reddit_3.4G', rel_path='Reddit_3.4G/')
+@register_task('reddit_13G', rel_path='Reddit_13G/')
+@register_task('reddit_softmax', rel_path='Reddit_2008/')
+class RedditTask(RankingTask):
```

Comment:
are these all really ranking tasks? These could use cleanup + more informative names, e.g. what is `reddit` vs `reddit_softmax`.

Reply:
@pappagari might know?

Reply:
Yeah all of them are ranking tasks. All of those tasks optimize binary cross entropy except reddit_softmax. In reddit_softmax, softmax is taken over all the samples in the batch (which includes one positive sample and batch_size-1 negative samples). And likelihood for the positive sample is maximized.

Reply:
Gotcha. Might rename some of them in cleaning up tasks but seems okay to leave it for this PR